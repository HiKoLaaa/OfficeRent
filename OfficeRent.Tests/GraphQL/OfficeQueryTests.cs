using System.Threading.Tasks;
using HotChocolate.Execution;
using NUnit.Framework;
using OfficeRent.Tests.Database.TestBuilders.Offices;
using OfficeRent.Tests.GraphQL;
using Snapshooter.NUnit;

namespace OfficeRent.Tests
{
	[TestFixture]
	public class OfficeQueryTests : GraphQLTestsBase
	{
		[Test]
		public async Task GetOffices_RequestThrowGraphQL_AllOfficesGot()
		{
			var office1 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Moscow").WithStreet("Koroleva").WithStreetNumber("1").Build())
				.WithName("office number 1")
				.WithFloor(1)
				.Build();

			var office2 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Abakan").WithStreet("Krilova").WithStreetNumber("2").Build())
				.WithName("office number 2")
				.WithFloor(7)
				.Build();

			await AddEntitiesToDbAsync(office1, office2);

			var executor = await ServiceProvider.GetRequestExecutorAsync();

			var request = QueryRequestBuilder
				.New()
				.SetQuery(OfficeQueries.GetAllOffices())
				.SetServices(ServiceProvider)
				.Create();

			var result = await executor.ExecuteAsync(request);

			result.MatchSnapshot();
		}

		[Test]
		public async Task GetOffice_RequestThrowGraphQL_OfficeGot()
		{
			var office1 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Moscow").WithStreet("Koroleva").WithStreetNumber("1").Build())
				.WithName("office number 1")
				.WithFloor(1)
				.Build();

			var office2 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Abakan").WithStreet("Krilova").WithStreetNumber("2").Build())
				.WithName("office number 2")
				.WithFloor(7)
				.Build();

			await AddEntitiesToDbAsync(office1, office2);

			var executor = await ServiceProvider.GetRequestExecutorAsync();

			var request = QueryRequestBuilder
				.New()
				.SetQuery(OfficeQueries.GetOffice(office1.Id))
				.SetServices(ServiceProvider)
				.Create();

			var result = await executor.ExecuteAsync(request);

			result.MatchSnapshot();
		}

		private static class OfficeQueries
		{
			public static string GetAllOffices()
			{
				return 
					@"query
{
	offices
	{
		id,
		name,
		address 
		{
			city,
			street,
			streetNumber
		},
		floor
		}
}";
			}

			public static string GetOffice<TId>(TId id)
			{
				return 
					$@"query
{{
	office(id: {id})
	{{
		id,
		name,
		address 
		{{
			city,
			street,
			streetNumber
		}},
		floor
		}}
}}";
			}
		}
	}
}