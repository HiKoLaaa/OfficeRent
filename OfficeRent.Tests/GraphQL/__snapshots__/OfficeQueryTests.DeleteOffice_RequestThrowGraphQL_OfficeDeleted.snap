{
  "Label": null,
  "Path": null,
  "Data": null,
  "Errors": [
    {
      "Message": "Entity of type Office with id 1 not found.",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "office"
      },
      "Locations": [
        {
          "Line": 3,
          "Column": 2
        }
      ],
      "Extensions": null,
      "Exception": {
        "EntityType": "OfficeRent.Api.Models.Offices.Office, OfficeRent.Api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",
        "EntityId": 1,
        "StackTrace": "   at OfficeRent.Api.Database.Abstractions.OfficeRepository.GetOfficeAsync(Int32 id) in D:\\RiderProjects\\OfficeRent\\OfficeRent.Api\\Database\\Abstractions\\Office\\OfficeRepository.cs:line 21\n   at OfficeRent.Api.GraphQL.Query.GetOfficeAsync(Int32 id, IOfficeRepository officeRepository) in D:\\RiderProjects\\OfficeRent\\OfficeRent.Api\\GraphQL\\Queries\\Query.cs:line 13\n   at HotChocolate.Resolvers.Expressions.ExpressionHelper.AwaitTaskHelper[T](Task`1 task)\n   at HotChocolate.Types.FieldMiddlewareCompiler.<>c__DisplayClass3_0.<<CreateResolverMiddleware>b__0>d.MoveNext()\n--- End of stack trace from previous location ---\n   at HotChocolate.Execution.Processing.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)\n   at HotChocolate.Execution.Processing.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)",
        "Message": "Entity of type Office with id 1 not found.",
        "Data": {},
        "InnerException": null,
        "HelpLink": null,
        "Source": "OfficeRent.Api",
        "HResult": -2146233088
      },
      "SyntaxNode": null
    }
  ],
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
